web/src/components/tasks/business-formation/review/section/ReviewSection.tsx

```diff
@@ -6,11 +6,13 @@ import { scrollToTop } from "@/lib/utils/helpers";
 import { PropsWithChildren, useContext } from "react";
 
 interface Props {
-  buttonText: string;
+  buttonText?: string;
   handleButtonClick?: () => void;
   header: string;
   stepName: FormationStepNames;
-  testId: string;
+  testId?: string;
+  sectionType: "MAIN" | "SUBSECTION";
+  marginOverride?: string;
```

Comment:
Correct me if I'm wrong, but here's what it seems:
- Every "MAIN" component has a `buttonText` and no `marginOverride`
- Every "SUBSECTION" component has no `buttonText` and a `margin-y-4` override.

There's maybe one exception I could find (`combinedInvestment` is a subsection that's missing the margin. But I looked at the design and that seems to be an error, it looks like it _should_ have the margin).

This kind of clear delineation between 2 cases, where one case uses one set of props exclusively, and another case uses another set of props exclusively, looks to me like a component that is begging to actually be 2 components. If you want to reduce duplication, you can make use of a shared (abstract) base component, but that's optional and might be more trouble than its worth. 

Basically what I'm saying is, if it would never make sense to have SUBSECTION that has an edit button (and from the designs, it looks like that's the case) then the component should not support that as a configurable option. #makeImpossibleStatesImpossible 

Following this suggestion will make our components more readable, because they won't be full of ternary logic and instead be declarative. Here's what a subsection looks like. Here's what a Main section looks like. No need to juggle so many class names and have a developer struggle to figure out what it'll actually render as.

Reply:
Also, two side notes, in terms of reducing unnecessary props.
- It looks like every time we pass a value to `buttonText`, it's the same Config editText. Why is it a prop if it's always the same?
- It looks like `handleButtonClick` is never used. It's an optional prop and there is no usage with it ever being passed a value. Why do we need it?